pkg/agent/tunnelserver/tunnelserver.go

```diff
@@ -195,6 +198,41 @@ func (t *tunnelServer) Log(ctx context.Context, message *tunnel.LogMessage) (*tu
 	return &tunnel.Empty{}, nil
 }
 
+func (t *tunnelServer) GitCloneAndRead(response *tunnel.Empty, stream tunnel.Tunnel_GitCloneAndReadServer) error {
+	if t.workspace == nil {
+		return fmt.Errorf("workspace is nil")
+	}
+
+	if t.workspace.Source.GitRepository == "" {
+		return fmt.Errorf("invalid repository")
+	}
+
+	gitCloneDir := filepath.Join(t.workspace.Folder, "source")
+
+	gitContext, cancel := context.WithTimeout(context.Background(), time.Minute)
```

Comment:
What if big repository that takes longer than a minute? I think we probably should not time this as we don't do it for regular reads as well